Add metadata for JGit 7.4.0

Our [Cloud config server smoke test](https://github.com/spring-projects/spring-aot-smoke-tests/tree/main/cloud/cloud-config-server) fails because of missing JGit metadata.

We've worked around that by using `buildArgs.addAll('-H:+UnlockExperimentalVMOptions', '-H:Preserve=package=org.eclipse.jgit.*')`.

We're using `org.eclipse.jgit:org.eclipse.jgit:7.4.0.202509020913-r`.

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'defaultEnvironmentRepository': Enumerated values of type org.eclipse.jgit.lib.CoreConfig$TrustStat not available
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1818)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:604)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:526)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:371)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:331)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:196)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1218)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1184)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1121)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:983)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:620)
	at org.springframework.boot.web.server.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:765)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:454)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1374)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363)
	at com.example.cloud.config.server.ConfigServerApplication.main(ConfigServerApplication.java:31)
	at java.base@25/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.IllegalArgumentException: Enumerated values of type org.eclipse.jgit.lib.CoreConfig$TrustStat not available
	at org.eclipse.jgit.lib.Config.allValuesOf(Config.java:585)
	at org.eclipse.jgit.lib.Config.getEnum(Config.java:572)
	at org.eclipse.jgit.lib.CoreConfig.inheritParseTrustStat(CoreConfig.java:319)
	at org.eclipse.jgit.lib.CoreConfig.parseTrustPackedRefsStat(CoreConfig.java:299)
	at org.eclipse.jgit.lib.CoreConfig.<init>(CoreConfig.java:269)
	at org.eclipse.jgit.lib.Config.get(Config.java:894)
	at org.eclipse.jgit.internal.storage.file.RefDirectory.<init>(RefDirectory.java:215)
	at org.eclipse.jgit.internal.storage.file.FileRepository.<init>(FileRepository.java:188)
	at org.eclipse.jgit.lib.BaseRepositoryBuilder.build(BaseRepositoryBuilder.java:679)
	at org.eclipse.jgit.api.InitCommand.call(InitCommand.java:104)
	at org.eclipse.jgit.api.CloneCommand.init(CloneCommand.java:279)
	at org.eclipse.jgit.api.CloneCommand.call(CloneCommand.java:187)
	at org.springframework.cloud.config.server.environment.JGitEnvironmentRepository.cloneToBasedir(JGitEnvironmentRepository.java:676)
	at org.springframework.cloud.config.server.environment.JGitEnvironmentRepository.initClonedRepository(JGitEnvironmentRepository.java:379)
	at org.springframework.cloud.config.server.environment.JGitEnvironmentRepository.afterPropertiesSet(JGitEnvironmentRepository.java:300)
	at org.springframework.cloud.config.server.environment.MultipleJGitEnvironmentRepository.afterPropertiesSet(MultipleJGitEnvironmentRepository.java:72)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1865)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1814)
	... 19 more
Caused by: java.lang.NoSuchMethodException: org.eclipse.jgit.lib.CoreConfig$TrustStat.values()
	at java.base@25/java.lang.Class.checkMethod(DynamicHub.java:1343)
	at java.base@25/java.lang.Class.getMethod(DynamicHub.java:1337)
	at org.eclipse.jgit.lib.Config.allValuesOf(Config.java:580)
	... 36 more
```

--- a/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/index.json
+++ b/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/index.json
@@ -1,0 +1,4 @@
+[
+  "reflect-config.json",
+  "resource-config.json"
+]

--- a/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/reflect-config.json
+++ b/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/reflect-config.json
@@ -1,0 +1,205 @@
+[
+  {
+    "name": "org.eclipse.jgit.internal.JGitText",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.nls.GlobalBundleCache"
+    },
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.internal.JGitText",
+    "allPublicFields": true,
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.nls.TranslationBundle"
+    }
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$AutoCRLF",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.treewalk.WorkingTreeOptions"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$CheckStat",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.treewalk.WorkingTreeOptions"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$EOL",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.treewalk.WorkingTreeOptions"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$HideDotFiles",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.internal.storage.file.FileRepository"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$HideDotFiles",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.treewalk.WorkingTreeOptions"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$LogRefUpdates",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.internal.storage.file.ReflogWriter"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$LogRefUpdates",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.lib.CoreConfig"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$SymLinks",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.treewalk.WorkingTreeOptions"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$TrustPackedRefsStat",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.internal.storage.file.RefDirectory"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.util.sha1.SHA1$Sha1Implementation",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.util.sha1.SHA1"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.CoreConfig$TrustStat",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.lib.Config"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.dircache.DirCache$DirCacheVersion",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.lib.Config"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  },
+  {
+    "name": "org.eclipse.jgit.lib.GpgConfig$GpgFormat",
+    "condition": {
+      "typeReachable": "org.eclipse.jgit.lib.Config"
+    },
+    "methods": [
+      {
+        "name": "values",
+        "parameterTypes": [
+
+        ]
+      }
+    ]
+  }
+]

--- a/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/resource-config.json
+++ b/metadata/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/resource-config.json
@@ -1,0 +1,7 @@
+{
+  "bundles": [
+    {
+      "name": "org.eclipse.jgit.internal.JGitText"
+    }
+  ]
+}

--- a/metadata/org.eclipse.jgit/org.eclipse.jgit/index.json
+++ b/metadata/org.eclipse.jgit/org.eclipse.jgit/index.json
@@ -1,8 +1,16 @@
 [
   {
     "latest": true,
+    "metadata-version": "7.4.0.202509020913-r",
+    "module": "org.eclipse.jgit:org.eclipse.jgit",
+    "tested-versions": [
+      "7.4.0.202509020913-r"
+    ]
+  },
+  {
     "metadata-version": "6.5.0.202303070854-r",
     "module": "org.eclipse.jgit:org.eclipse.jgit",
+    "default-for" : "6\\..*",
     "tested-versions": [
       "6.5.0.202303070854-r"
     ]

--- a/tests/src/index.json
+++ b/tests/src/index.json
@@ -425,6 +425,12 @@
     "versions" : [ "6.5.0.202303070854-r" ]
   } ]
 }, {
+  "test-project-path" : "org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r",
+  "libraries" : [ {
+    "name" : "org.eclipse.jgit:org.eclipse.jgit",
+    "versions" : [ "7.4.0.202509020913-r" ]
+  } ]
+}, {
   "test-project-path" : "org.eclipse.paho/org.eclipse.paho.client.mqttv3/1.2.5",
   "libraries" : [ {
     "name" : "org.eclipse.paho:org.eclipse.paho.client.mqttv3",

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/.gitignore
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/build.gradle
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/build.gradle
@@ -1,0 +1,30 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.eclipse.jgit:org.eclipse.jgit:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'ch.qos.logback:logback-classic:1.4.5'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "metadata-user-code-filter.json"
+                extraFilterPath = "metadata-extra-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/gradle.properties
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/gradle.properties
@@ -1,0 +1,2 @@
+library.version = 7.4.0.202509020913-r
+metadata.dir = org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/metadata-extra-filter.json
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/metadata-extra-filter.json
@@ -1,0 +1,5 @@
+{
+  "rules": [
+    {"includeClasses": "org.eclipse.jgit.**"}
+  ]
+}

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/metadata-user-code-filter.json
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/metadata-user-code-filter.json
@@ -1,0 +1,5 @@
+{
+  "rules": [
+    {"includeClasses": "org.eclipse.jgit.**"}
+  ]
+}

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/settings.gradle
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/settings.gradle
@@ -1,0 +1,13 @@
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.eclipse.jgit.org.eclipse.jgit_tests'

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/java/org_eclipse_jgit/org_eclipse_jgit/OrgEclipseJGitTest.java
@@ -1,0 +1,119 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jgit.org_eclipse_jgit;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.internal.storage.file.FileRepository;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Fail.fail;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class OrgEclipseJGitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger("OrgEclipseJGitTest");
+
+    private static final String TMP_DIR = System.getProperty("java.io.tmpdir");
+
+    private static final String TEST_FILE_NAME = "Test.txt";
+
+    private static final String GIT_DIR_NAME = ".git";
+
+    private File remoteRepositoryDir;
+
+    private Repository remoteRepository;
+
+    private Git remoteGit;
+
+    @BeforeAll
+    void beforeAll() throws Exception {
+        remoteRepositoryDir = new File(TMP_DIR, "test_jgit_remote_repo_" + System.currentTimeMillis());
+        remoteRepository = new FileRepository(new File(remoteRepositoryDir, GIT_DIR_NAME));
+        remoteRepository.create();
+        logger.info("Created remote repository: {}", remoteRepositoryDir);
+        remoteGit = new Git(remoteRepository);
+        writeFile(remoteRepository, TEST_FILE_NAME, "Test Message");
+        remoteGit.add().addFilepattern(TEST_FILE_NAME).call();
+        RevCommit revCommit = remoteGit.commit().setMessage("Initial commit").call();
+        logger.info("Committed '{}' file to remote repository: {}", TEST_FILE_NAME, revCommit.getName());
+    }
+
+    private void writeFile(Repository repository, String name, String data) throws IOException {
+        File parentFile = repository.getWorkTree();
+        File file = new File(parentFile, name);
+        try (Writer w = new OutputStreamWriter(new FileOutputStream(file), StandardCharsets.UTF_8)) {
+            w.write(data);
+        }
+    }
+
+    @AfterAll
+    void afterAll() {
+        if (remoteRepository != null) {
+            remoteRepository.close();
+        }
+        deleteDir(remoteRepositoryDir);
+    }
+
+    @Test
+    void test() throws Exception {
+        File localRepositoryDir = new File(TMP_DIR, "test_jgit_local_repo_" + System.currentTimeMillis());
+        try {
+            Git localGit = Git.cloneRepository()
+                    .setDirectory(localRepositoryDir)
+                    .setURI("file://" + remoteGit.getRepository().getWorkTree().getPath())
+                    .call();
+            logger.info("Created local repository: {}", localRepositoryDir);
+
+            File[] localFiles = localRepositoryDir.listFiles();
+            if (localFiles != null) {
+                logger.info("Found files in local repository: {}", Arrays.toString(localFiles));
+                assertThat(localFiles)
+                        .hasSize(2)
+                        .extracting(File::getName)
+                        .contains(TEST_FILE_NAME, GIT_DIR_NAME);
+            } else {
+                fail("Files not found in local repository");
+            }
+
+            localGit.getRepository().close();
+        } finally {
+            deleteDir(localRepositoryDir);
+        }
+    }
+
+    private void deleteDir(File file) {
+        if (file != null) {
+            File[] childFiles = file.listFiles();
+            if (childFiles != null) {
+                for (File childFile : childFiles) {
+                    deleteDir(childFile);
+                }
+            }
+            boolean deleted = file.delete();
+            if (!deleted) {
+                logger.warn("File '{}' not deleted", file);
+            }
+        }
+    }
+
+}

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/resources/META-INF/native-image/jgit-tests-only/resource-config.json
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/resources/META-INF/native-image/jgit-tests-only/resource-config.json
@@ -1,0 +1,12 @@
+{
+  "resources": {
+    "includes": [
+      {
+        "pattern": "\\Qlogback.xml\\E",
+        "condition": {
+          "typeReachable": "ch.qos.logback.core.util.Loader"
+        }
+      }
+    ]
+  }
+}

--- a/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/resources/logback.xml
+++ b/tests/src/org.eclipse.jgit/org.eclipse.jgit/7.4.0.202509020913-r/src/test/resources/logback.xml
@@ -1,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <logger name="OrgEclipseJGitTest" level="INFO"/>
+
+    <root level="ERROR">
+        <appender-ref ref="STDOUT" />
+    </root>
+
+</configuration>

